tidy3d/components/monitor.py

```diff
@@ -184,6 +184,17 @@ def set_default_colocate(cls, val, values):
             val = not sum(interval_space) == 3
         return val
 
+    def get_downsampled_num_cells(self, num_cells: Tuple[int, int, int]) -> Tuple[int, int, int]:
+        """Given a tuple of the number of cells spanned by the monitor along each dimension,
+        return the number of cells one would have after downsampling based on ``interval_space``.
+        """
+        num_cells_downsampled = list(num_cells.copy())
+        for idx, interval in enumerate(self.interval_space):
+            if interval > 1:
+                num_cells_downsampled[idx] = np.floor((num_cells[idx] - 1) / interval) + 1
+                num_cells_downsampled[idx] += int((num_cells[idx] - 1) % interval > 0)
+        return num_cells_downsampled
```

Comment:
This function is different than `downsample_size` used in `test_monitors` on the backend. Could you check which one is correct, and use the AbstractFieldMonitor method in the test?